Services/Certificate/classes/Placeholder/Values/class.ilDefaultPlaceholderValues.php

```diff
@@ -7,63 +7,23 @@
  */
 class ilDefaultPlaceholderValues implements ilCertificatePlaceholderValues
 {
-    /**
-     * @var array
-     */
-    private $placeholder;
-
-    /**
-     * @var ilCertificateObjectHelper
-     */
-    private $objectHelper;
-
-    /**
-     * @var ilCertificateDateHelper
-     */
-    private $dateHelper;
-
-    /**
-     * @var integer
-     */
-    private $dateFormat;
-
-    /**
-     * @var ilLanguage|null
-     */
-    private $language;
-
-    /**
-     * @var ilCertificateUtilHelper|null
-     */
-    private $utilHelper;
+    private array $placeholder;
+    private ?ilCertificateObjectHelper $objectHelper;
+    private ?ilCertificateDateHelper $dateHelper;
+    private ?int $dateFormat;
+    private ?ilLanguage $language;
+    private ?ilCertificateUtilHelper $utilHelper;
+    private ?ilUserDefinedFieldsPlaceholderValues $userDefinedFieldsPlaceholderValues;
```

Comment:
These types are not optional. Only the passed constructor arguments are optional.